design/handle-backup-of-volumes-by-resources-filters.md

```diff
@@ -0,0 +1,223 @@
+# Handle backup of volumes by resources filters
+ 
+## Abstract
+Currently, Velero doesn't have one flexible way to filter volumes.
+ 
+If users want to skip backup of volumes or only backup some volumes in different namespaces in batch, currently they need to use the opt-in and opt-out approach one by one, or use label-selector but if it has big different labels on each different related pods, which is cumbersome when they have lots of volumes to handle with. it would be convenient if Velero could provide one way to filter the backup of volumes just by `some specific volumes attributes`.
+ 
+Also, currently, it's not accurate enough if the users want to select a specific volume to do a backup or skip by without patching labels or annotations to the pods. It would be useful if users could accurately select target volume by `one specific resource selector`. Users could accurately select the volume to backup or skip in their own console when using velero for secondary development.
+ 
+## Background
+As of Today, Velero has lots of filters to handle (backup or skip backup) resources including resources filters like `IncludedNamespaces, ExcludedNamespaces`, label selectors like `LabelSelector, OrLabelSelectors`, annotation like `backup.velero.io/must-include-additional-items` etc. But it's not enough flexible to handle volumes, we need one generic way to filter volumes.
+ 
+## Goals
+- Introducing one flexible way to filter volumes.
```

Comment:
This mechanism should have the flexibility to allow we decide the actual action taken to the pv satisfy a certain filter, for example for PVs provisioned via storage class A, the user wants to take snapshots, for PVs provisioned by storage class B, the user wants to bakcup via kopia.

We don't need to support that in the first version of implementation, but we should make sure there's such extensibility.

Reply:
which means that through the resource filter configuration for one **single** backup, Velero could both take snapshots for some volumes and  Kopia/Restic backup for other volumes?

Reply:
In our backup workflow, we list PVs and decide whether to take snapshots or not, we list pods and find volumes mount in pods and decide whether to backup or not. which is two different code path.

Maybe we should consider how to relate pod volumes to PV attributes when using PV attributes as filters under Kopia/Restic backup